src/st/no3/4CA54.c

```diff
@@ -0,0 +1,1068 @@
+    distance = GetDistanceToPlayerX();
+    if ((distance < 0x48) && (entity->step != 4)) {
+        SetStep(4);
+        return;
+    }
+    SetStep(3);
+    if (distance < 0x60) {
+        entity->ext.et_801CC820.unk7C = 1;
+    } else {
+        entity->ext.et_801CC820.unk7C = 0;
+    }
+    entity->ext.et_801CC820.unk80 = 0;
+    entity->ext.et_801CC820.unk82 = 0x20;
+}
+
+void func_801CC90C(Entity* arg0) {
+    s16 temp_v0_2;
+    s16 temp_v1;
+    s16 temp_v0;
+
+    temp_v0 = GetDistanceToPlayerX();
+    temp_v1 = arg0->ext.generic.unk84.S16.unk2;
+    if (temp_v1 != 0) {
+
+        if ((u32)(temp_v0) < 0x60) {
+            temp_v0_2 = temp_v1 - 2;
+            arg0->ext.generic.unk84.S16.unk2 = temp_v0_2;
+            if (temp_v0_2 < 0) {
+                arg0->ext.generic.unk84.S16.unk2 = 0;
+            }
+        } else {
+            arg0->ext.generic.unk84.S16.unk2 = (temp_v1 - 1);
+        }
+    }
+}
+
+void EntityStrongWarg(Entity* self) {
+    Entity* ent_s0;
+    Entity* ent_s4;
+    EnemyDef* enemyDefPtr;
+    s16* hitboxPtr;
+    Collider collider;
+    u16 var_s1;
+    s32 var_s2;
+    u8 frameIdx;
+
+    var_s1 = self->step;
+    if (self->flags & 0x100) {
+        if (!self->params) {
+            if (var_s1 != 0xB) {
+                self->velocityX = 0;
+                self->velocityY = 0;
+                SetStep(0xB);
+            }
+        } else {
+            ent_s0 = (self + 1);
+            self->params = 0;
+            self->flags &= ~(FLAG_NOT_AN_ENEMY | FLAG_DEAD);
+            ent_s0->flags &= ~(FLAG_NOT_AN_ENEMY | FLAG_DEAD);
+            self->flags |= FLAG_UNK_400000 | FLAG_UNK_400;
+            ent_s0->flags |= FLAG_UNK_400000;
+            self->enemyId = ent_s0->enemyId = 0x94;
+            ent_s0->flags = self->flags;
+            enemyDefPtr = &g_api.enemyDefs[148];
+            self->hitPoints = enemyDefPtr->hitPoints;
+            ent_s0->attack = self->attack;
+            ent_s0->attack = self->attack;
+            ent_s0->unk60 = self;
+
+            for (var_s2 = 0; var_s2 < 11; var_s2++) {
+                self->unk6D[var_s2] = 0x10;
+                ent_s0->unk6D[var_s2] = 0x10;
+            }
+
+            if (var_s1 != 4) {
+                SetStep(4);
+                SetSubStep(1);
+                self->velocityY = FIX(-5.0);
+                if (self->facingLeft) {
+                    self->velocityX = FIX(-1.5);
+                } else {
+                    self->velocityX = FIX(1.5);
+                }
+            }
+        }
+    } else if (self->hitFlags & 0xF) {
+        if (self->ext.factory.unk86 >= 0x10) {
+            self->ext.factory.unk86 /= 2;
+        }
+        if ((var_s1 == 2 || var_s1 == 3) && !self->params) {
+            SetStep(0xA);
+        }
+    }
+
+    var_s1 = self->step;
+    switch (var_s1) {
+    case 0:
+        ent_s0 = self + 1;
+        self->unk60 = ent_s0;
+        // PSP version: 0x20
+        CreateEntityFromCurrentEntity(0x30, ent_s0);
```

Comment:
`E_ID_30` https://github.com/Xeeynamo/sotn-decomp/blob/95930a2fe303e0e00063af80ff8c07421cda7bbd/src/st/no3/no3.h#L50

You could also add `E_ID_31` for line 189 below